pkg/controller/resourcedistribution/utils.go

```diff
@@ -302,3 +297,34 @@ func listNamespacesForDistributor(handlerClient client.Client, targets *appsv1al
 
 	return matchedSet.List(), unmatchedSet.List(), nil
 }
+
+func needToUpdate(old, new *unstructured.Unstructured) bool {
+	if !reflect.DeepEqual(old.GetAnnotations(), new.GetAnnotations()) {
```

Comment:
what if other components update the secrets & configmap with additional annotations& labels? can we simply check other fields